etc/CMakeLists.txt

```diff
@@ -0,0 +1,3 @@
+# Install the bash completion script in an unversioned location.
```

Comment:
This makes it harder to install that `gz-tools1` and `gz-tools2` side-by-side. They're currently conflicting because they both install the `ign` executable, but once we migrate `gz-tools2` to the new executable (maybe `gz`), they shouldn't conflict anymore.

This is all to say, can we install this in a versioned directory?

Reply:
Hmm it might be a little awkward. Currently, we said we want to installing all the libraries' scripts to the same directory, `gz/gz.completion.d`, so that we can source a single script `gz/gz.completion`, which sources all the scripts in that directory.

The version will probably need to go on the `gz` directory. But what version should it be? If we make it the version of `tools`, then when the upper libraries install to the same directory, they need to `find_package()` on `tools` to get its version, right? Does that introduce extra dependency? Looks like that isn't an explicit dependency right now.

Reply:
Good point about the added dependency. @j-rivero has been changing `ign_find_package(ignition-tools QUIET)` to `find_program(HAVE_IGN_TOOLS ign)` in PRs like https://github.com/gazebosim/gz-gui/pull/240.

If we really want to avoid the `find_package` calls, maybe we can just hardcode the version on downstream libraries like `set(GZ_TOOLS_VER 1)` 🤔 That's not much different from hardcoding the version number in the package name itself. It's just a bit more error prone, since nothing on CI would indicate to us when that number is wrong.

Reply:
I think that might be doable right now. I'll try it.

Another point came up though. I was writing tests for ign-plugin, and it occurred to me that the available flags are slightly different between ign-plugin1 and ign-plugin2. Similarly for ign-transport8 and ign-transport10 - the new one has `--json-output`, the old one doesn't.

That gives us some complication about where to install the scripts. If we install them to a directory versioned by the `ign-tools` version, Citadel and Fortress both have `ign-tools1`, so they would install to the same directory, say `install/share/gz/gz1.completion.d`.

But the upper libraries have different versions, and there could be different flags, like the example above. So maybe we have a `gui3` file for Citadel, a `gui6` file for Fortress. But we can't install both to the directory versioned by ign-tools, `install/share/gz/gz1.completion.d`, because all the scripts in that directory will get sourced, and the two `_gz_gui` functions will conflict.

We can't name the function names by the versions, like `_gz_gui3`, `gz_gui6`, because that means the bash script here in ign-tools will need to know the library version to call the right function.

Coming back to the original problem, to distinguish between versions (transport8 and transport10 say), they'll need different versions of the bash completion script in the source code (that means manual forward-port. Or we could skip the unimportant flags for now, and just not include them in the bash completion, for automatic forward-port to work). To install them so that different versions don't get sourced and conflict, they will need to be installed in different `*.completion.d` directories, and those directories can't be versioned by ign-tools. They will need to be versioned by the individual libraries. But... then what, I'm stuck in the thought process here. We wouldn't know which directory to source from a bash script... that script would need to know all those library numbers, wouldn't it?

Reply:
Ahh ok. Some thoughts:

* When the user has 2 versions of a command installed side-by-side, I think it's reasonable to autocomplete based on the newest version. Just like we default to the newest version unless you explicitly `--force-version` a lower version.

* `gz-tools` has ways of finding out all the versions of a command that are currently available, you can try for example `ign gui --versions`. So it could use this information in `etc/ign.bash_completion.sh` to assemble the `_gz_$cmd$version` function name.

* Or much easier, rely on the fact that bash allows redefining a function, and that files are iterated in increasing order, to just redefine the `_gz_gui` function on multiple files. This way the definition of the highest version should always prevail. I just tried this for example:

```
#!/usr/bin/env bash
function _gz_gui
{
  echo "A"
}
function _gz_gui
{
  echo "B"
}
_gz_gui
# Prints B
```

Reply:
I think it's working now. Going with bullet 3 above per offline discussions.

6b39016 installs to a versioned location, `share/gz#/gz#.completion*`.
I updated the ign-plugin and ign-gui PRs accordingly. They install to `share/gz#/gz#.completion.d/<lib>#.bash_completion.sh`.

I'm testing different versions in the same workspace by manually copying these scripts to higher versions locally.
The installation of these particular scripts worked:
```
$ colcon build --merge-install --packages-select ignition-gui3 ignition-gui6 ignition-gui7 ignition-plugin1 ignition-plugin2 ignition-tools ignition-tools2

$ ls install/share/gz/
gz.completion  gz1.completion.d  gz2.completion.d

$ ls install/share/gz/gz1.completion.d/ -1
gui3.bash_completion.sh
gui6.bash_completion.sh
gui7.bash_completion.sh
ign1.bash_completion.sh
plugin1.bash_completion.sh
plugin2.bash_completion.sh

$ ls install/share/gz/gz2.completion.d/
ign2.bash_completion.sh
```

Test it:
```
. install/share/gz/gz.completion
ign gui -<tab>
ign plugin -<tab>
```

Since I just copied the scripts locally, you can't tell which version is in use.
You can add some extra output to check it. For example, I added to `install/share/gz/gz1.completion.d/gui6.bash_completion.sh` an extra flag `--66666666666666`, and to the `gui7` file an extra `--777777777777`.
The higher version is indeed overwriting the lower version:
```
$ . install/share/gz/gz.completion
$ ign gui -
--777777777777   --force-version  --list           --verbose        -c               -l               -v               
--config         --help           --standalone     --versions       -h               -s               
```

Reply:
Thanks! It looks like `install/share/gz/gz.completion` will still collide between the versions. Can that be `install/share/gz/gz1.completion `?

Reply:
d879208 does the above - install the script to versioned name but still sources all gz*.completion.d directories.

Reply:
Thanks, that works for me!